templates/tfengine/components/cicd/triggers/plan/main.tf

```diff
@@ -0,0 +1,118 @@
+# limitations under the License.
+
+resource "google_cloudbuild_trigger" "plan_env" {
+  count       = var.skip ? 0 : 1
+  disabled    = var.run_on_push
+  provider    = google-beta
+  project     = var.project_id
+  name        = "tf-plan-${var.env}"
+  description = "Terraform plan job triggered on push event."
+
+  included_files = [
+    "${var.terraform_root_prefix}**",
+  ]
+
+  {{if has $ "github" -}}
+  github {
+    owner = var.github.owner
+    name  = var.github.name
+    pull_request {
+      branch = "^${var.branch_name}$"
+    }
+  }
+  {{- else if has $ "cloud_source_repository" -}}
+  trigger_template {
+    repo_name   = var.cloud_source_repository.name
+    branch_name = "^${var.branch_name}$"
+  }
+  {{- end}}
+
+  filename = "${var.terraform_root_prefix}cicd/configs/tf-plan.yaml"
+
+  substitutions = {
+    _TERRAFORM_ROOT = var.terraform_root
+    _MANAGED_DIRS = var.managed_dirs
+  }
+
+  depends_on = [
+    google_project_service.services,
+{{- if has $ "cloud_source_repository"}}
+    google_sourcerepo_repository.configs,
+{{- end}}
+  ]
+}
+
+# Create another trigger as Pull Request Cloud Build triggers cannot be used by Cloud Scheduler.
+resource "google_cloudbuild_trigger" "plan_scheduled_env" {
+  count       = (!var.skip && var.run_on_schedule != "") ? 1 : 0
+  # Always disabled on push to branch.
+  disabled    = true
+  provider    = google-beta
+  project     = var.project_id
+  name        = "tf-plan-scheduled-${var.env}"
+  description = "Terraform plan job triggered on schedule."
+
+  included_files = [
+    "${var.terraform_root_prefix}**",
+  ]
+
+  {{if has $ "github" -}}
+  github {
+    owner = var.github.owner
+    name  = var.github.name
+    push {
+      branch = "^${var.branch_name}$"
+    }
+  }
+  {{- else if has $ "cloud_source_repository" -}}
+  trigger_template {
+    repo_name   = var.cloud_source_repository.name
+    branch_name = "^${var.branch_name}$"
+  }
+  {{- end}}
+
+  filename = "${var.terraform_root_prefix}cicd/configs/tf-plan.yaml"
+
+  substitutions = {
+    _TERRAFORM_ROOT = var.terraform_root
+    _MANAGED_DIRS = var.managed_dirs
+  }
+
+  depends_on = [
+    google_project_service.services,
+{{- if has $ "cloud_source_repository"}}
+    google_sourcerepo_repository.configs,
+{{- end}}
+  ]
+}
+
+resource "google_cloud_scheduler_job" "plan_scheduler_env" {
+  count     = (!var.skip && var.run_on_schedule != "") ? 1 : 0
+  project   = var.project_id
+  name      = "plan-scheduler-${var.env}"
+  region    = var.scheduler_region
+  schedule  = var.run_on_schedule
+  time_zone = "America/New_York" # Eastern Standard Time (EST)
+  attempt_deadline = "60s"
+  http_target {
+    http_method = "POST"
+    oauth_token {
+      scope = "https://www.googleapis.com/auth/cloud-platform"
+      service_account_email = "${google_service_account.cloudbuild_scheduler_sa.email}"
+    }
+    uri = "https://cloudbuild.googleapis.com/v1/${google_cloudbuild_trigger.plan_scheduled_env}.id}:run"
```

Comment:
I think there's a typo here with `}.id}`

Reply:
Not sure, [this is the original](https://github.com/GoogleCloudPlatform/healthcare-data-protection-suite/blob/master/templates/tfengine/components/cicd/triggers.tf#L250)

I'll do a deployment, it surely pop out if it's wrong

Reply:
The original is correct - it has one set of `${...}` and inside it `{{...}}`. This one just one set of `${...}` and an extra `}` right before `.id`.

Reply:
I see. Actually the typo was the `}` before.

Thanks